pydrive2/files.py

```diff
@@ -344,20 +349,33 @@ def download(fd, request):
                 download(fd, files.get_media(fileId=file_id))
             except errors.HttpError as error:
                 exc = ApiRequestError(error)
-                if (
-                    exc.error["code"] != 403
-                    or exc.GetField("reason") != "fileNotDownloadable"
-                ):
+                code = exc.error["code"]
+                reason = exc.GetField("reason")
+                if code == 403 and reason == "fileNotDownloadable":
+                    mimetype = mimetype or "text/plain"
+                    fd.seek(0)  # just in case `download()` modified `fd`
+                    try:
+                        download(
+                            fd,
+                            files.export_media(
+                                fileId=file_id, mimeType=mimetype
+                            ),
+                        )
+                    except errors.HttpError as error:
+                        raise ApiRequestError(error)
+                elif code == 416 and reason == "requestedRangeNotSatisfiable":
+                    # NOTE: An empty file case. Wasting one API call to make
+                    # absolutely sure. See
+                    # https://github.com/iterative/dvc/issues/4507
+                    try:
+                        req = files.get_media(fileId=file_id)
```

Comment:
does it actually download the file?

I think we can do `self.FetchMetadata(fields="size")` (need to check the field name "size") + `self.metadata.get("size")`?

Reply:
Yes, the `execute()` does download the whole contents, but this only happens when we are verifying that the file is empty and otherwise we'll error-out. Sure, could do the `size`. Thanks.